utils/docker/docker_entrypoint.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+set -e
+
+cd $HOME/self_balancing_robot
+
+catkin config --extend /opt/ros/melodic --cmake-args -DCMAKE_BUILD_TYPE=RelWithDebInfo
+catkin init && catkin build
+
+echo "source $HOME/self_balancing_robot/devel/setup.bash" >> $HOME/.bashrc
+echo "source $HOME/self_balancing_robot/src/utils/scripts/setup.bash" >> $HOME/.bashrc
```

Comment:
I think maybe I'm not understanding entrypoints correctly - this also only should happen once when setting up the container, so this can also move to Dockerfile? Or is there something preventing these one-time setup commands from being executed as `RUN` commands in the Dockerfile?

(But I think I am wrong - please let me know!)

Reply:
not this one; since it requires the files mounted from the host.
You don't have access to those in the Dockerfile, only after the container starts with the mounts.